packages/backend/src/backend_factory.ts

```diff
@@ -108,6 +115,32 @@ export class BackendFactory<
   createStack = (name: string): Stack => {
     return this.stackResolver.createCustomStack(name);
   };
+
+  setCustomOutput = (
+    key: string,
+    value: string,
+    options?: CustomOutputOptions
+  ): void => {
+    /*
+    Looks like this must be in certain format.
+    From attempt to use non-alpha numeric characters with this.outputStorageStrategy.addBackendOutputEntry() in early prototype.
+
+    The CloudFormation deployment has failed. Find more information in the CloudFormation AWS Console for this stack.
+    Caused By: ❌ Deployment failed: Error [ValidationError]: Template format error: Outputs name 'amplify-custom-myApiUrl-value' is non alphanumeric.
+     */
+
+    if (!this.customOutputsEntry) {
+      this.customOutputsEntry =
+        this.outputStorageStrategy.addAppendableBackendOutputEntry(
+          'AWS::Amplify::Custom'
+        );
+    }
+
+    const outputData = JSON.stringify({ value: value, options });
+    console.log(outputData);
+    // TODO how do we do custom namespacing ??
+    this.customOutputsEntry.appendOutput(`amplifycustom${key}`, outputData);
+  };
```

Comment:
1. It seems that we can't just use output key to smuggle information about destination client config path due to CFN validation.
2. To overcome that I just put serialized json with output value AND metadata that describes where this should go in client config.
3. Is versioning of `AWS::Amplify::Custom` good enough to deal with these serialized jsons?

Reply:
3. It should be. If we change anything about the format within the "AWS::Amplify::Custom" block we'd need to increment that version. That would cover any changes to those serialized JSON objects.